api/CSVHandler.py

```diff
@@ -50,3 +48,44 @@ def test_split_train_test():
     train, test = split_train_test(data)
     assert np.shape(train) == (40, 4)
     assert np.shape(test) == (27, 4)
+
+def create_graph_nodes(data):
+    for line in data:
+        if (line[0].isdigit()) and ("->" not in line):
+            node = line.split("\"")[1].split("\\n")
+            node_id = int(line[0])
+            gini = ""
+            samples = ""
+            values = ""
+            expression = ""
+            for attr in node:
+                if 'gini' in attr:
+                    gini = float(attr.split(" ")[-1])
+                elif 'samples' in attr:
+                    samples = float(attr.split(" ")[-1])
+                elif 'value' in attr:
+                    values = attr.split("[")[-1].replace("]", "").replace(" ", "").split(",")
+                    values = list(map(int, values))
+                else:
+                    expression = attr
+            
+            with driver.session() as session:
+                if expression:
+                    session.write_transaction(create_rule_node, node_id, expression, gini, samples, values)
+                else:
+                    session.write_transaction(create_leaf_node, node_id, gini, samples, values)
+
+def create_rule_node(tx, identifier, expression, gini, samples, value):
+    tx.run("MERGE (a:Rule {identifier: $identifier, expression: $expression, gini: $gini, samples: $samples, value: $value}) ",
```

Comment:
Again, if you aren't sure whether you'll get all of these params passed through (I think it's safe to say that we will), this becomes a more interesting problem. We might have to pass a default `None` so that when you do your `$` interpolation the Neo4j driver will know there's nothing there.